lib/tlscontext.h

```diff
@@ -64,8 +65,21 @@ typedef struct _TLSSession
   TLSSessionVerifyFunc verify_func;
   gpointer verify_data;
   GDestroyNotify verify_data_destroy;
+  TLSPeerInformation *peer_info;
 } TLSSession;
```

Comment:
I think we dont need to use a separate pointer here, just embed the struct here.